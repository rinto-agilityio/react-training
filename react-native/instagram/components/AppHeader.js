import React from "react";
import { View, Text, StyleSheet } from "react-native";

export default class AppHeader extends React.Component {
  render() {
    return (
      <View style={styles.wrapper}>
        <Text style={styles.brand}>Instagram</Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  wrapper: {
    width: "100%",
    paddingTop: 20,
    paddingBottom: 10,
    backgroundColor: "#f9f9f9",
    justifyContent: "center",
    alignItems: "center",
    borderBottomWidth: 1,
    borderColor: "#eee"
  },
  brand: {
    fontSize: 24
  }
});
