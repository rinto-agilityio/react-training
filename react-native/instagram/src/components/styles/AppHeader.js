import { StyleSheet } from "react-native";

export const styles = StyleSheet.create({
  wrapper: {
    width: "100%",
    paddingTop: 20,
    paddingBottom: 10,
    backgroundColor: "#f9f9f9",
    justifyContent: "center",
    alignItems: "center",
    borderBottomWidth: 1,
    borderColor: "#eee"
  },
  brand: {
    fontSize: 24
  }
});
