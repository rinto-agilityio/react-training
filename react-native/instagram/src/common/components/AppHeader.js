// Libs
import React from 'react'
import { View, Text } from 'react-native'

// Helpers
import { APP_NAME } from '@constants'

// Styles
import { styles } from './styles/AppHeaderStyles'

const AppHeader = () => (
  <View style={styles.wrapper}>
    <Text style={styles.brand}>{APP_NAME}</Text>
  </View>
)

export default AppHeader
