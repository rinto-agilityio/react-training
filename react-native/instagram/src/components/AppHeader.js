import React from 'react';
import { View, Text } from 'react-native';
import { styles } from './styles/AppHeader';

export default class AppHeader extends React.Component {
  render() {
    return (
      <View style={styles.wrapper}>
        <Text style={styles.brand}>Instagram</Text>
      </View>
    );
  }
}
