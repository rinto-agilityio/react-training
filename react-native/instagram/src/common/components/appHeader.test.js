import AppHeader from './AppHeader'

describe('AppHeader component', () => {
  it('AppHeader renders correctly', () => {
    const treeDOM = renderer.create(<AppHeader />).toJSON()

    expect(treeDOM).toMatchSnapshot()
  })
})
