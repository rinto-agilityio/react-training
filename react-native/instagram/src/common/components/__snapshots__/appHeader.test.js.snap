// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader component AppHeader renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#f9f9f9",
      "borderBottomWidth": 1,
      "borderColor": "#eee",
      "justifyContent": "center",
      "paddingBottom": 10,
      "paddingTop": 20,
      "width": "100%",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 24,
      }
    }
  >
    Instagram
  </Text>
</View>
`;

exports[`AppHeader component Renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#f9f9f9",
      "borderBottomWidth": 1,
      "borderColor": "#eee",
      "justifyContent": "center",
      "paddingBottom": 10,
      "paddingTop": 20,
      "width": "100%",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 24,
      }
    }
  >
    Instagram
  </Text>
</View>
`;
